lessons/python/inttopics/inttopics.ipynb

```diff
@@ -0,0 +1,672 @@
+    "- the interpreter\n",
+    "- variables\n",
+    "- lists\n",
+    "- indexing/slicing\n",
+    "- if statements\n",
+    "- for loops (and loop syntax in general)\n",
+    "- functions\n",
+    "- very basic NumPy, Pandas, matplotlib"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reading (and Writing) Files\n",
+    "\n",
+    "Let's say we have 50 students who wrote a test. This test was out of 40 points, and their results - in percentages - are stored in a text file called `scores.txt`. We want to import these into a Python list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# let's make sure of our working directory\n",
+    "\n",
+    "import os\n",
+    "os.chdir('/Users/Ahmed/Desktop') "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "with open('scores.txt', 'r') as file:\n",
+    "    scores = file.read().split(',') "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "print(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The eagle-eyed amongst you may notice the quotes around each number - but we'll address that in a bit."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "## Methods and Attributes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "len(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our txt file missed a student! Let's say she got an 89 on the test. How can we add her score to our list?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# should we do this?\n",
+    "\n",
+    "scoresfixed = scores + [89]  \n",
+    "\n",
+    "# or should we open the file in Notepad/TextEdit and manually put the 89 in?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Python contains built-in methods and attributes for each object type\n",
+    "dir(scores) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "help(scores.append) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks like what we want, but let's check on `insert` just to be sure -"
```

Comment:
Was not aware of that! I've only ever added one thing at a time when appending, but that is handy to know. I'll update the notebook to reflect this as well.